frontend/src/components/LoginForm/LoginForm.js

```diff
@@ -1,5 +1,159 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { 
+  Button,
+  CardContent,
+  Checkbox,
+  IconButton,
+  InputAdornment,
+  Grid,
+  FormControlLabel,
+  Link,
+  Paper,
+  TextField,
+  Typography
+} from '@material-ui/core';
+import visibilityIcon from '../../assets/generic/visibility.svg';
+import visibilityOffIcon from '../../assets/generic/visibility-off.svg';
+import productIcon from "../../assets/generic/product-icon.svg"
+import { emailRegex, passwordRegex } from "../../constants/regex"
 
-const LoginForm = () => <h2>I&apos;m a login form, look at me</h2>;
+export default function LoginForm(props) {
+  const [email, setEmail] = useState("")
+  const [password, setPassword] = useState("")
+  const [rememberMe, setRememberMe] = useState(false)
+  const [showPassword, setShowPassword] = useState(false)
+  const [showError, setShowError] = useState({ errorEmail: false, errorPassword: false })
 
-export default LoginForm;
+  const handleLogin = (event) => {
+    event.preventDefault()
+    if (!emailRegex.test(email)) {
+      setShowError({ ...showError, errorEmail: true })
+      return
+    }
+    if (!passwordRegex.test(password)) {
+      setShowError({ ...showError, errorPassword: true })
+      return
+    }
+    // [TM]: TODO WDB-13
+    props.onLogin(email, password)
+  }
+
+  const handleEmailChange = (event) => {
+    setEmail(event.target.value)
+    setShowError({ ...showError, errorEmail: false })
+  }
+
+  const handlePasswordChange = (event) => {
+    setPassword(event.target.value)
+    setShowError({ ...showError, errorPassword: false })
+  }
+
+  const handleRememberMeCheck = (event) => {
+    // [TM]: TODO WDB-16
+    setRememberMe(event.target.checked)
+  }
+
+  const handleClickShowPassword = () => setShowPassword(!showPassword)
+  const handleMouseDownPassword = (event) => event.preventDefault()
+
+  return (
+    <Paper elevation={3}>
+      <CardContent
+        direction="column"
+        align="center"
+        justify="center"
+      >
+        <img
+          src={productIcon}
+          alt="Video library logo"
+          width={50}
+          height={50}
+        />
+        <Typography gutterBottom variant="h4">Login</Typography>
+        <form noValidate onSubmit={handleLogin}>
```

Comment:
expand on your decision to use noValidate here. First time I see that

Reply:
I believe this is because @tmaciulis22 is ditching default validation and providing his own. This should make sure that no duplicate actions occur,

Reply:
@kibartas  Default validation provides default tooltips about what is wrong. In our case we are using custom messages with styling provided by Material UI